src/System.Management.Automation/engine/Utils.cs

```diff
@@ -990,7 +990,32 @@ internal static bool NativeItemExists(string path, out bool isDirectory, out Exc
             if (result == -1)
             {
                 int errorCode = Marshal.GetLastWin32Error();
-                if (errorCode == 5)
+
+                // Errorcode 32 is 'ERROR_SHARING_VIOLATION' i.e. 
+                // The process cannot access the file because it is being used by another process.
+                // GetFileAttributes will return INVALID_FILE_ATTRIBUTES for a system file or directory because of this error.
+                // Fallback to FindFirstFile to check if the file actually exists or not.
+                if (errorCode == 32)
+                {
+                    IntPtr INVALID_HANDLE_VALUE = new IntPtr(-1);
+                    NativeMethods.WIN32_FIND_DATA findData;
+                    IntPtr findHandle;
+                    findHandle = NativeMethods.FindFirstFile(path, out findData);
```

Comment:
GetFileAttributes can get the handle to a file even if it is shared by another process but it fails when the sharing flag for the file is set to 0x00000000. This happen because GetFileAttributes function tries to open the file with FILE_READ_ATTRIBUTES access right. It prevents us from opening a file for delete, read, or write access. In current scenario, C:\pagefile.sys is always opened by OS with sharing flag 0x00000000. But FindFirstFile is still able to get attributes as this api retrieves the required information using a find handle generated with FILE_LIST_DIRECTORY access.